api/HouseInput.cs

```diff
@@ -1,23 +1,93 @@
-﻿/**
+/**
  * Reading device state from the house app.
  */
 using System;
+using System.Threading;
+using System.IO;
+using System.Diagnostics;
+using Newtonsoft.Json;
+using System.Net;
+using System.Threading.Tasks;
+using System.Text;
 
 namespace api
 {
-/**
- * Class which reads state from the house app.
- */
-public class HouseInput : IDeviceInput
-{
-	public HouseInput()
-	{
-	}
+    /**
+     * Class which reads state from the house app.
+     */
+    public class HouseInput : IDeviceInput
+    {
+        public HouseInput(string URL)
+        {
+            _URL = URL;
+        }
 
-	public bool read(Device dev)
+        public bool read(Device dev)
 	{
-		return true;
+		var _ = readHelper(dev);
+
+        return true;
 	}
-}
+
+        private async Task<string> readHelper(Device dev)
+        {
+            try
+            {
+                var request = (HttpWebRequest)WebRequest.Create(_URL);
+                request.Method = "GET";
+                request.ContentType = "application/json";
+
+                try
+                {
+                    using (var stream = await Task<Stream>.Factory.FromAsync(request.BeginGetRequestStream, request.EndGetRequestStream, request))
+                    {
+                        result = new byte[stream.Length];
+                        await stream.ReadAsync(result, 0, (int)stream.Length);
+                    }
+
+                    //Interfaces.UpdateDevice(dev, result.ToString());
+                    
+                    // TODO: use DeviceIDOnlyConverter
+                }
+                catch (Exception ex)
+                {
+                    _StreamException = ex;
+                    return null;
+                }
+                
+            }
+            catch (Exception ex)
+            {
+                _URLException = ex;
+                return null;
+            }
```

Comment:
Also, for this block, have you considered trying out RestSharp, or some other REST API library? I haven't counted lines, but it seems from my quick testing that it may reduce verbosity dramatically.